sagemaker/modelmonitor.ipynb

```diff
@@ -0,0 +1,413 @@
+   "source": [
+    "%store -r\n",
+    "import os,json\n",
+    "from dkube.sdk import *\n",
+    "from dkube.sdk.api import DkubeApi\n",
+    "from dkube.sdk.rsrcs import operator\n",
+    "from dkube.sdk.rsrcs.operator import DkubeCluster\n",
+    "from dkube.sdk.rsrcs import DkubeModelmonitor\n",
+    "from dkube.sdk.rsrcs.modelmonitor import DatasetClass,ModelType,DriftAlgo\n",
+    "from dkube.sdk.rsrcs.modelmonitor import DatasetFormat,DkubeModelmonitoralert"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api = DkubeApi(token=os.getenv(\"DKUBE_USER_ACCESS_TOKEN\"))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Predict Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "PREDICT_PREFIX = PREFIX +\"/datacapture/\"+endpoint_name+\"/AllTraffic/\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pname =  MONITOR_NAME+'-predict'\n",
+    "\n",
+    "if DATA_SOURCE == 'aws_s3':\n",
+    "    try:\n",
+    "        dataset = DkubeDataset(DKUBEUSERNAME, name=pname,remote=True)\n",
+    "        dataset.update_dataset_source('aws_s3')\n",
+    "        dataset.update_awss3_details(\n",
+    "            bucket=BUCKET,\n",
+    "            prefix=PREDICT_PREFIX,key=os.getenv(\"AWS_ACCESS_KEY_ID\",ACCESS_KEY),\n",
+    "            secret=os.getenv(\"AWS_SECRET_ACCESS_KEY\",SECRET_KEY))\n",
+    "        api.create_dataset(dataset)\n",
+    "        \n",
+    "    except Exception as e:\n",
+    "        if e.reason:\n",
+    "            if e.reason.lower() != \"conflict\":\n",
+    "                response = e.body\n",
+    "                print(f\"Failed[{response.code}]: {response.message}\")\n",
+    "        else:\n",
+    "            raise e"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Configure Sagemaker Cluster in Dkube"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "pcluster = DkubeCluster(name=SAGEMAKER_DKUBE_CLUSTER_NAME)\n",
+    "pcluster.update_kind(\"sagemaker\")\n",
+    "pcluster.update_access_keys(ACCESS_KEY,SECRET_KEY)\n",
+    "pcluster.cluster.access_keys.region = REGION_NAME"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api.configure_clusters(pcluster.cluster)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Import Deployment"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "api.import_deployment(name=MONITOR_NAME,cluster=SAGEMAKER_DKUBE_CLUSTER_NAME,variant=\"sagemaker\")"
```

Comment:
This should return the ID of the deployment. Please pass the ID as well when you create modelmonitor below. Update all the examples as well

Reply:
done in PR https://github.com/oneconvergence/dkube/pull/103